src/tests/ftest/nvme/enospace.py

```diff
@@ -0,0 +1,440 @@
+
+        #Fill 60% more of SCM pool, now NVMe will be Full so data will not be
+        #moved to NVMe but it will start filling SCM. SCM size will be going to
+        #full and this command expected to fail with DER_NOSPACE
+        try:
+            self.start_ior_load(storage='SCM', precent=60)
+            self.fail('This test suppose to FAIL because of DER_NOSPACE'
+                      'but it got Passed')
+        except TestFail as _error:
+            self.log.info('Test expected to fail because of DER_NOSPACE')
+
+        #Display the pool%
+        print(self.pool.pool_percentage_used())
+
+        #verify the DER_NO_SAPCE error count is expected and no other Error in
+        #client log
+        self.verify_enspace_log(self.der_nospace_count)
+
+        #Check both NVMe and SCM are full.
+        pool_usage = self.pool.pool_percentage_used()
+        #NVM should be 99% full if not test will fail.
+        if pool_usage['nvme'] > 1:
+            self.fail('Pool NVMe used percentage should be < 1%, instead {}'.
+                      format(pool_usage['nvme']))
+        #For SCM some % space used for system so it wont be 100% full.
+        if pool_usage['scm'] > 25:
+            self.fail('Pool SCM used percentage should be < 20%, instead {}'.
+                      format(pool_usage['scm']))
+
+    def run_enospace_with_bg_job(self):
+        """
+        Function to run test and validate DER_ENOSPACE and expected storage
+        size. Single IOR job will run in background while space is filling.
+        """
+        #Get the initial DER_ENOSPACE count
+        self.der_enspace_log_count()
+
+        # Start the IOR Background thread which will write small data set and
+        # read in loop, until storage space is full.
+        out_queue = queue.Queue()
+        job = threading.Thread(target=self.ior_bg_thread,
+                               kwargs={"results": out_queue})
+        job.daemon = True
+        job.start()
+
+        #Run IOR in Foreground
+        self.run_enospace_foreground()
+        # Verify the background job queue and make sure no FAIL for any IOR run
+        while not self.out_queue.empty():
+            if self.out_queue.get() == "FAIL":
+                self.fail("One of the Background IOR job failed")
+
+    @skipForTicket("DAOS-4846")
+    def test_enospace_lazy_with_bg(self):
+        """Jira ID: DAOS-4756.
+
+        Test Description: IO gets DER_NOSPACE when SCM and NVMe is full with
+                          default (lazy) Aggregation mode.
+
+        Use Case: This tests will create the pool and fill 75% of SCM size which
+                  will trigger the aggregation because of space pressure,
+                  next fill 75% more which should fill NVMe. Try to fill 60%
+                  more and now SCM size will be full too.
+                  verify that last IO fails with DER_NOSPACE and SCM/NVMe pool
+                  capacity is full.

```

Comment:
Actually aggregation will be triggered when SCM is 60% full (in lazy mode), or system in idle (no io activities) for 5 seconds.

Reply:
Yes, I am just putting 75% to overshoot the 60%